### ISSUE TITLE
QDialog opens at the center of QGoodWindow's initial position, ignoring its current one

### ISSUE THREAD

**Issue by kresimirko** [antonypro/QGoodWindow#23]
Hello. I'm back again with another issue and this time I really am not sure if this is an issue on my behalf or not.

When using a regular QMainWindow, connecting a QAction to a QDialog's `open()` slot works fine. The dialog's parent is the QMainWindow and it opens in the center, even after moving.

```c++
// Example code
AboutWindow *aboutWindow = new AboutWindow(this); // inherits QDialog
connect(helpAction, &QAction::triggered, aboutWindow, &AboutWindow::show);
```

![image](https://user-images.githubusercontent.com/104278083/213736034-9ba9d977-d5e7-4e78-b9f7-7dec8db86e47.png)
_Initial position_

![image](https://user-images.githubusercontent.com/104278083/213736063-a25ead58-8fcd-4f17-9c5c-6f1cbc4a6281.png)
_Position after being moved by cursor_

However, if I connect that same QAction to the same `show()` slot, but this time have a QGoodWindow as its parent, this happens:

![image](https://user-images.githubusercontent.com/104278083/213740571-08bd8d13-c452-402d-a440-7ca768b6a71c.png)
_Initial position_

![image](https://user-images.githubusercontent.com/104278083/213740596-a7530a27-525f-467d-91e0-12d1e1fd4db8.png)
_Position after being moved by cursor_

If I use `move()` on the dialog's parent window before it being shown, that is used as the initial position instead.

**Comment by kresimirko** [comment]
I'm aware of the fact that I can get the position of the window on the screen, calculate things from there and position the dialog that way, but I still feel like this is notable.

**Comment by kresimirko** [comment]
For a real example, see this:

https://github.com/kresimirko/neoazul/blob/8f44a6202514dd70d7c256cf41b668aa28f10e9a/neoazul/main.cpp#L133 (things get initialized here; if `-qgw` is passed the program uses QGoodWindow with a normal QMainWindow as a child, otherwise that same QMainWindow is launched standalone)
https://github.com/kresimirko/neoazul/blob/8f44a6202514dd70d7c256cf41b668aa28f10e9a/neoazul/widgets/centraluiwindow.cpp#L53 (normal QMainWindow's QAction connection, if it's running standalone)
https://github.com/kresimirko/neoazul/blob/8f44a6202514dd70d7c256cf41b668aa28f10e9a/neoazul/widgets/mainwindow.cpp#L143 (QGoodWindow's QAction connection, notice the commented out lambda below which enables correct positioning)

Ask me if You need more explanations about this source code.
Please comment out the file writing thing in `main.cpp` if you want to build and run this.

**Comment by antonypro** [comment]
@kresimirko: I don't think that this is exactly a bug, because the library works, but not mimic the `QMainWindow` behavior.

As a workaround, you could, instead of create a global `QDialog` and show and close it as user request, create a temporary `QDialog` every time the dialog is requested, this also ensure that your dialog will have a Windows show animation on any show event, and not remember it's old position on show event.

Anyway, I will accept your issue as a enhancement to be released in the next version, it will mimic the `QMainWindow` behavior.

Thanks for reporting!

**Comment by kresimirko** [comment]
Alright, thanks! I've already applied a workaround so I'll just stick to that.

**Comment by antonypro** [comment]
This enhancement was addressed on version 2.2.
